lightning/src/ln/channelmanager.rs

```diff
@@ -6938,11 +7034,14 @@ where
 			htlc_purposes.push(purpose);
 		}
 
-		(per_peer_state.len() as u64).write(writer)?;
+		(peers_to_searialize_count).write(writer)?;
 		for (peer_pubkey, peer_state_mutex) in per_peer_state.iter() {
-			peer_pubkey.write(writer)?;
-			let peer_state = peer_state_mutex.lock().unwrap();
-			peer_state.latest_features.write(writer)?;
+			let peer_state_lock= peer_state_mutex.lock().unwrap();
+			let peer_state = & *peer_state_lock;
+			if !peer_state.ok_to_remove(false) {
```

Comment:
I think a little comment explaining why we're now only serializing the peers that are not safely removable might help, though I guess it's pretty clear.